### ISSUE TITLE
Incorrect order of the checkout fields when using locale

### ISSUE THREAD

**Issue by pierrebuet** [woocommerce/woocommerce#21751]
On the checkout page and when locale are active (for example french or german)  the display of the input fields is weird and not in the order the locale would have triggered in the previous version (for example postcode and city ). 

I have already a working patch to correct this, but since I still have one pull request  which is approved but not merged, I don't know if I should pull the correction for this bug now or wait for the first pull request to be merged.

**Comment by BusinessBloomer** [comment]
Yes, having the same problem. Filter works (woocommerce_default_address_fields) but only when you change billing country.

An example: 

```
add_filter( 'woocommerce_default_address_fields', 'bbloomer_reorder_checkout_fields' );
 
function bbloomer_reorder_checkout_fields( $fields ) {
  $fields['company']['priority'] = 8;
  return $fields;
}
```

This used to work, but now it only does when you change country and not on load. Source: [https://businessbloomer.com/woocommerce-move-reorder-fields-checkout-page/](https://businessbloomer.com/woocommerce-move-reorder-fields-checkout-page/)

Let us know :)

**Comment by pierrebuet** [comment]
So I just made a pull request with explanation and the solution, but this got merged with the previous pull request, sorry about that.

**Comment by kloon** [comment]
This is fixed by https://github.com/woocommerce/woocommerce/pull/21733

**Comment by pierrebuet** [comment]
@kloon : I'm having trouble to understand where you trigger the address reordering on checkout. That said, I just want to remember you that the culprit was the commit deleting

$( '#billing_country, #shipping_country, .country_to_state' ).change(); 

from checkout.js. #billing_country and #shipping_country will be either a select or an hidden field if you allow only one country in your setting.

That said, my solution would have been to eliminate the trigger on form loading as wanted in the initial commit who deleted the line and deliver the fields directly in the right order, eliminating eventual FOUC on page load, for example.

**Comment by kloon** [comment]
I have refactored the priority sorting in a new PR #21763, you were on the right track but it was not working for all the filters that could change the fields.

Closing as it gets resolved by #21763
